cmd/rm.go

```diff
@@ -37,25 +34,49 @@ func rm(cmd *cobra.Command, args []string) (err error) {
 		return
 	}
 	if pathMetaData.Tag == folder {
-		return fmt.Errorf("rm: cannot remove ‘%s’: Is a directory", path)
+		if !recursive {
+			return fmt.Errorf("rm: cannot remove ‘%s’: Is a directory", path)
+		}
+		entries, err := ListFiles(path)
+		if err != nil {
+			return err
+		}
+		for _, e := range entries {
+			switch e.Tag {
+			case "folder":
+				removePath(pathmod.Join(path, e.Folder.Name), recursive)
+			case "file":
+				removePath(pathmod.Join(path, e.File.Name), recursive)
+			}
+		}
 	}
 
 	arg := files.NewDeleteArg(path)
 
 	if _, err = dbx.Delete(arg); err != nil {
 		return
 	}
+	return nil
+}
 
-	return
+func rm(cmd *cobra.Command, args []string) (err error) {
+	if len(args) != 1 {
+		return errors.New("`rm` requires a `file` or `folder` argument")
+	}
+	recursive, _ := cmd.Flags().GetBool("recursive")
+	return removePath(args[0], recursive)
 }
 
 // rmCmd represents the rm command
 var rmCmd = &cobra.Command{
 	Use:   "rm [flags] <file>",
 	Short: "Remove files",
-	RunE:  rm,
+	Example: `dbxcli rm <file>
+	dbxcli rm -r <folder>`,
```

Comment:
Hmmm, you're right that it will slow it down. Perhaps a good question to think about is: do we want the functionality to mirror that of the rm/rmdir of coreutils or align it more closely with the dropbox web interface. It can go either way, that's certainly your call @diwakergupta. 

If you want it to be closer to the web ui (which it seems like it is), As a user, I would like to see a check to say "This directory is not empty, are you sure you want to do this (y/n)" or requiring a force flag (-f/--force). Using dropbox's web interface has a prompt to confirm deletion so I think that's reasonable.